### ISSUE TITLE
Variable scope breaks when directly setting p5.Vector values

### ISSUE THREAD

**Issue by ccheaton** [processing/p5.js#739]
I recently started a project wherein I create an instance of a custom `Arc` class and then recursively create children, based on a value set as a global parameter. The `Arc` class has a method `Arc.prototype.draw`. That method looks like this:

```javascript
Arc.prototype.draw = function(){
  // Position child arcs
  if (this.is_root == false){
    this.set_position(); // NOTE THIS LINE
  }

  // Draw the arc
  strokeWeight(this.stroke_width);
  noFill();
  stroke(colors[this.color_index]);
  arc(this.center.x,this.center.y,this.diameter,this.diameter,this.degree_start,this.degree_stop);

  // Recursively draw the children arcs
  if (this.child_arc !== null){
    this.child_arc.draw();
  }
}
```

When I call `root_arc.draw()` in the main `draw()` loop, you'll note that it recursively draws the children arcs. For each child arc, it's position is first set with the `Arc.prototype.set_position` class method, which looks like this:

```javascript
Arc.prototype.set_position = function(){

  var vecB       = this.point_along_arc(this.parent_arc.center.x, this.parent_arc.center.y, this.parent_arc.diameter, this.attachment_angle);
  var vecA       = createVector(this.parent_arc.center.x,this.parent_arc.center.y);
  var new_center = p5.Vector.lerp(vecA,vecB,1- arc_diameter_decay);

  // Comment this out and uncomment the two lines below it and watch the difference in behavior
  this.center = createVector(new_center.x,new_center.y);

  // this.center.x  = new_center.x;
  // this.center.y  = new_center.y;
}
```

Notice the note about the change in behavior. Basically, if you activate the lines at the bottom and then comment out the line indicated, then the representation of `this` becomes global for the vector stored at `this.center`, the scoping rules are broken, and setting the the lines:

```javascript
this.center.x  = new_center.x;
this.center.y  = new_center.y;
```
changes not only the `x` and `y` values of the `center` instance variable of the child `Arc`, but the change bubbles up to the parent object, too.

In other words, it is unsafe to directly set the values in a `p5.Vector` in this type of situation because the expected scoping rules of JavaScript (as I understand them) are broken. Creating a new vector and assigning it to `this.center` does not cause a problem.

The easiest way to see this is to run the code, changing the commenting as indicated on line 109. Here it is!

https://github.com/ccheaton/p5js_sketches/blob/9f7554de74d234ac1423657d19c915f25ed6711c/golden/sketch.js

**Comment by brysonian** [comment]
The problem is that on line [92](https://github.com/ccheaton/p5js_sketches/blob/9f7554de74d234ac1423657d19c915f25ed6711c/golden/sketch.js#L92) you are passing the _center variable by reference. So all arcs share the same vector instance as their center. When you do createVector you set each center to a unique vector object.

If you want to set properties directly, you need to make a copy of the vector before passing to the constructor: 

```
this.child_arc = new Arc(_center.copy(),...
```

**Comment by ccheaton** [comment]
Ah - right. Thanks!
